state-chain/runtime/src/lib.rs

```diff
@@ -1735,7 +1735,8 @@ impl_runtime_apis! {
 			all_prewitnessed_swaps
 		}
 
-		fn cf_scheduled_swaps(base_asset: Asset, _quote_asset: Asset) -> Vec<(SwapLegInfo, BlockNumber)> {
+		fn cf_scheduled_swaps(base_asset: Asset, quote_asset: Asset) -> Vec<(SwapLegInfo, BlockNumber)> {
+			assert_eq!(quote_asset, STABLE_ASSET, "Only USDC is supported as quote asset");
```

Comment:
Why is this assert necessary? It's not necessarily always going to be the case that we'll only allow USDC as the quote asset.

Also not sure about panicking here anyway, or below on something that's totally possible since it's user input. Anyone could provide the wrong asset. I think we should throw an error here instead, that way the user can actually get the error in the response too, instead of "server error" or something

Reply:
> Why is this assert necessary?

Because the `get_scheduled_swap_legs` function (and entire swapping pallet) assumes the stable asset is usdc. I wanted to sanity check it here.

> totally possible since it's user input

The parent functions do check that the pool exists and return an error. Not exactly confirming that quote asset is usdc, but I am unsure how to get `cf_scheduled_swaps` to return an error.